src/ui/SearchInterface/ScrollRestoration.ts

```diff
@@ -0,0 +1,89 @@
+import { ResultListEvents, IDisplayedNewResultsEventArgs } from '../../events/ResultListEvents';
+import { $$ } from '../../utils/Dom';
+import { Logger } from '../../misc/Logger';
+import { StorageUtils } from '../../utils/StorageUtils';
+import { QueryStateModel } from '../../models/QueryStateModel';
+
+export interface IScrollRestorationInfo {
+  lastPosition: number;
+  pageHeight: number;
+}
+
+export class ScrollRestoration {
+  static ID = 'ScrollRestoration';
+
+  private scrollInfoStorage: StorageUtils<Record<string, IScrollRestorationInfo>>;
+
+  private restorationTimeOutInMs: number = 5000;
+  private tryToScrollIntervalInMs: number = 50;
+  private timeoutHandle = null;
+
+  constructor(public element: HTMLElement, private queryStateModel: QueryStateModel) {
```

Comment:
Same comment as previous PR: I'd rename `element` to `root`

Reply:
Sure